lib/analyse_ocamlformat.mli

```diff
@@ -1,2 +1,3 @@
+type version = Version of string | Vendored
 
-val get_ocamlformat_version : Current.Job.t -> Fpath.t -> string option Lwt.t
+val get_ocamlformat_version : opam_files:string list -> Current.Job.t -> Fpath.t -> version option Lwt.t
```

Comment:
This is an odd type signature now. Some suggestions for improvements:
 - keep `Job.t` first as this is distinct from the others;
 - shift `opam_files` after the `Fpath.t` argument to respect their relative dependency;
 - add comment to clarify that `opam_files` are file-paths relative to the root, not e.g. file contents or absolute paths.

```suggestion
val get_ocamlformat_version : Current.Job.t -> root:Fpath.t -> opam_files:string list -> version option Lwt.t
(* Get the OCamlformat version to use for linting a given project directory, given the locations of the [.opam] files in the repository relative to the project root *)
```

Reply:
I did most of that, what do you think?